starboard/shared/modular/starboard_layer_posix_socket_abi_wrappers.cc

```diff
@@ -77,27 +176,86 @@ SB_EXPORT int __abi_wrap_connect(int sockfd,
 
 SB_EXPORT int __abi_wrap_getaddrinfo(const char* node,
                                      const char* service,
-                                     const struct addrinfo* hints,
-                                     struct addrinfo** res) {
-  int result = getaddrinfo(node, service, hints, res);
+                                     const struct musl_addrinfo* hints,
+                                     struct musl_addrinfo** res) {
+  if (res == nullptr)
+    return -1;
+  // musl addrinfo definition might differ from platform definition.
+  // So we need to do a manual conversion to avoid header mismatches.
+  struct addrinfo new_hints = {0};
 
-#if SB_HAS_QUIRK(SOCKADDR_WITH_LENGTH)
-  struct addrinfo* ai = *res;
+  if (hints != nullptr) {
+    if (musl_hints_to_platform_hints(hints, &new_hints) == -1)
+      return -1;
+  }
+
+  addrinfo* native_res = nullptr;
+  int result = getaddrinfo(
+      node, service,
+      (hints == nullptr) ? nullptr
+                         : reinterpret_cast<const struct addrinfo*>(&new_hints),
+      &native_res);
+
+  struct addrinfo* ai = native_res;
+  struct musl_addrinfo* last_ai = nullptr;
+  *res = nullptr;
   while (ai != nullptr) {
     if (ai->ai_addr != nullptr) {
-      musl_sockaddr* musl_addr_ptr =
-          reinterpret_cast<musl_sockaddr*>(ai->ai_addr);
-      struct sockaddr* addr_ptr =
-          reinterpret_cast<struct sockaddr*>(ai->ai_addr);
-      uint8_t sa_family = addr_ptr->sa_family;
-      musl_addr_ptr->sa_family = sa_family;
+      addrinfo ai_copy = *ai;
+      struct musl_addrinfo* musl_ai =
+          (struct musl_addrinfo*)calloc(1, sizeof(struct musl_addrinfo));
+      if (platform_hints_to_musl_hints(&ai_copy, musl_ai) == -1) {
+        free(musl_ai);
+        return -1;
+      }
+      musl_ai->ai_addrlen = ai_copy.ai_addrlen;
+      musl_ai->ai_addr =
+          (struct musl_sockaddr*)calloc(1, sizeof(struct musl_sockaddr));
+      for (int i = 0; i < sizeof(PLATFORM_AF_ORDERED) / sizeof(int); i++) {
+        if (ai_copy.ai_addr->sa_family == PLATFORM_AF_ORDERED[i])
+          musl_ai->ai_addr->sa_family = MUSL_AF_ORDERED[i];
+      }
+      // Using sizeof() because sa_data is a static array sometimes.
+      if (sizeof(ai_copy.ai_addr->sa_data) > sizeof(nullptr)) {
```

Comment:
In previous github pre-submit, clang-3-9 compiler threw an error about `sa_data != nullptr` being pointless because sa_data is a static array and will never be null.